spec/datadog/profiling/crashtracker_spec.rb

```diff
@@ -0,0 +1,201 @@
+      before { crashtracker.start }
+      after { crashtracker.stop }
+
+      it 'starts a second crash tracker for the fork' do
+        expect_in_fork do
+          crashtracker.reset_after_fork
+
+          expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 2
+
+          crashtracker.stop
+
+          expect(`pgrep -f libdatadog-crashtracking-receiver`.lines.size).to be 1
+        end
+      end
+    end
+  end
+
+  describe '#stop' do
+    subject(:stop) { crashtracker.stop }
+
+    context 'when _native_stop_crashtracker raises an exception' do
+      it 'logs the exception' do
+        expect(described_class).to receive(:_native_stop) { raise 'Test failure' }
+        expect(Datadog.logger).to receive(:error).with(/Failed to stop crash tracking: Test failure/)
+
+        stop
+      end
+    end
+
+    it 'stops the crash tracker' do
+      crashtracker.start
+
+      stop
+
+      expect(`pgrep -f libdatadog-crashtracking-receiver`).to be_empty
+    end
+  end
+
+  context 'integration testing' do
+    shared_context 'HTTP server' do
+      let(:server) do
+        WEBrick::HTTPServer.new(
+          Port: 0,
+          Logger: log,
+          AccessLog: access_log,
+          StartCallback: -> { init_signal.push(1) }
+        )
+      end
+      let(:hostname) { '127.0.0.1' }
+      let(:log) { WEBrick::Log.new(StringIO.new, WEBrick::Log::WARN) }
+      let(:access_log_buffer) { StringIO.new }
+      let(:access_log) { [[access_log_buffer, WEBrick::AccessLog::COMBINED_LOG_FORMAT]] }
+      let(:server_proc) do
+        proc do |req, res|
+          messages << req.tap { req.body } # Read body, store message before socket closes.
+          res.body = '{}'
+        end
+      end
+      let(:init_signal) { Queue.new }
+
+      let(:messages) { [] }
+
+      before do
+        server.mount_proc('/', &server_proc)
+        @server_thread = Thread.new { server.start }
+        init_signal.pop
+      end
+
+      after do
+        unless RSpec.current_example.skipped?
+          # When the test is skipped, server has not been initialized and @server_thread would be nil; thus we only
+          # want to touch them when the test actually run, otherwise we would cause the server to start (incorrectly)
+          # and join to be called on a nil @server_thread
+          server.shutdown
+          @server_thread.join
+        end
+      end
+    end
+
+    include_context 'HTTP server'
+
+    let(:request) { messages.first }
+    let(:port) { server[:Port] }
+
+    let(:exporter_configuration) { [:agent, "http://#{hostname}:#{port}"] }
+
+    it 'reports crashes via http' do
+      fork_expectations = proc do |status:, stdout:, stderr:|
+        expect(Signal.signame(status.termsig)).to eq('SEGV').or eq('ABRT')
+        expect(stderr).to include('[BUG] Segmentation fault')
+      end
+
+      expect_in_fork(fork_expectations: fork_expectations) do
+        crashtracker.start
+
+        Process.kill('SEGV', Process.pid)
```

Comment:
Would it also make sense to test an actual crash (e.g. null ptr deref)

Reply:
That I know of, there's no Ruby-level way to cause a segfault. I even thought to check the built-in ffi fiddle gem, which actually is neat enough to handle its own potential segfaults cleanly and turn them into Ruby exceptions:

> spec/datadog/profiling/crashtracker_spec.rb:188:in `[]': NULL pointer dereference (Fiddle::DLError)

So to actually do this, I would need to add a `cause_crash` method on the native side of the Profiler, which.... seems like a sharp edge that is not great to leave lying around >_> ?

Reply:
You didn't try hard enough? :D

```
ruby -r fiddle -e 'Fiddle.free(42)'
-e:1: [BUG] Segmentation fault at 0x0000000000000022
```

Reply:
Ah, interesting, I didn't think of that :D . Thanks @lloeki for the suggestion, I'll make a note to shoot a small PR with your suggestion :)